docs/template_plugin/tests/functional/op_reference/tensor_iterator.cpp

```diff
@@ -0,0 +1,2306 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "openvino/op/tensor_iterator.hpp"
+#include "base_reference_test.hpp"
+#include <ngraph/op/util/attr_types.hpp>
+#include "ngraph_functions/builders.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct TensorIteratorParams {
+    template <class T>
+    TensorIteratorParams(
+        const size_t batchSize, const size_t inputSize, const size_t hiddenSize, const size_t seqLength,
+        const float clip, const ngraph::op::RecurrentSequenceDirection& direction,
+        const ngraph::helpers::TensorIteratorBody& body_type,
+        const element::Type_t& iType,
+        const std::vector<T>& XValues, const std::vector<T>& H_tValues,  const std::vector<T>& C_tValues, const std::vector<int64_t>& S_tValues,
+        const std::vector<T>& WValues, const std::vector<T>& RValues, const std::vector<T>& BValues,
+        const std::vector<T>& YValues, const std::vector<T>& HoValues, const std::vector<T>& CoValues,
+        const std::string& testcaseName = "") :
+        batchSize(batchSize), inputSize(inputSize), hiddenSize(hiddenSize), seqLength(seqLength),
+        clip(clip), body_type(body_type), direction(direction), iType(iType), oType(iType),
+        testcaseName(testcaseName) {
+            switch (body_type) {
+                case ngraph::helpers::TensorIteratorBody::LSTM: {
```

Comment:
Could you use enums and classes from ov namespace?

Reply:
Fixed it! I used local enum for only this file. Thanks!